### PR TITLE
feat: dark mode now works for login/signup pages too

### PR DESCRIPTION
**Pull request by okatu-loli**
Fix: https://github.com/casdoor/casdoor/issues/3233

### CODE REVIEW

**Comment by hsluoyz** [comment]
@dacongda plz review

**Comment by hsluoyz** [comment]
@dacongda plz review

File: web/src/App.js [hunk 1]
```diff
@@ -446,7 +448,7 @@ class App extends Component {
                 setLogoutState={() => {
                   this.setState({
                     account: null,
-                    themeAlgorithm: ["default"],
+                    // themeAlgorithm: ["default"],
```
Comment (dacongda) [inline web/src/App.js:451]: remove comment if it is unnecessary

File: web/src/App.js [hunk 2]
```diff
@@ -277,6 +277,7 @@ class App extends Component {
         <Footer id="footer" style={
           {
             textAlign: "center",
+            background: this.state.themeAlgorithm.includes("dark") ? "black" : "white",
```
Comment (dacongda) [inline web/src/App.js:280]: footer will auto adjust, we should never control the background style in code

File: web/src/auth/LoginPage.js [hunk 3]
```diff
@@ -1283,7 +1283,7 @@ class LoginPage extends React.Component {
             <div className="side-image" style={{display: application.formOffset !== 4 ? "none" : null}}>
               <div dangerouslySetInnerHTML={{__html: application.formSideHtml}} />
             </div>
-            <div className="login-form">
+            <div className="login-form" style={this.props.themeAlgorithm && this.props.themeAlgorithm.includes("dark") ? {backgroundColor: "black"} : {}}>
```
Comment (dacongda) [inline web/src/auth/LoginPage.js:1286]: control background in style will override customCss

**Comment by dacongda** [comment]
and i suggest you to provide a demo video or screenshot

**Comment by okatu-loli** [comment]
this is demo video

https://github.com/user-attachments/assets/36946143-bc6e-40f3-9ac7-6cd34d52907d

**Review by dacongda — APPROVED** [review]

**Pull request merged**